ergo-wallet/src/main/java/org/ergoplatform/wallet/interface4j/SecretString.java

```diff
@@ -13,40 +18,66 @@
      */
     private final char[] _data;
 
+    /**
+     * Erased flag, should not be copied outside of this instance.
+     */
+    private boolean _erased;
+
     /**
      * Use static methods to construct new instances.
      */
     SecretString(char[] data) {
         _data = data;
+        _erased = false;
     }
 
     /**
      * Returns true if the string doesn't have characters.
      */
-    public boolean isEmpty() { return _data == null || _data.length == 0; }
+    public boolean isEmpty() {
+        if (_erased) {
+            throw new RuntimeException("SecretString already erased");
+        } else {
+            return _data == null || _data.length == 0;
+        }
+    }
 
     /**
      * Extracts secret characters as an array.
+     * Throws an exception if {@link SecretString#_erased} flag is true
      */
     public char[] getData() {
-        return _data;
+        if (_erased) {
+            throw new RuntimeException("SecretString already erased");
```

Comment:
Throwing the same exception from all methods should be extracted into a single method with some meaningful name.

Reply:
What's the point of extracting the throw? It muddies the code readability in my opinion.
If the reason is to be sure the text of the error thrown is the same, it could be extracted to a const.

Reply:
The point is that once error condition is detected, some _logical unit of work_ has to be done. 
Throwing is part of this unit of work, now happen to be the only part, but something else can be done as well, like collecting some data. 
You can also think of the SecretString to move kind of error state (yes you don't save this state now, but anyway).
It is better to have all these encapsulated.

The readability depends on the name you give the method.

Reply:
Added method, please check